### PR TITLE
Convert `AdderToolbar` to TypeScript

### PR DESCRIPTION
**Pull request by lyzadanger**
Well, this was quite straightforward...

Depends on #4753

### CODE REVIEW

File: src/annotator/components/AdderToolbar.tsx [hunk 1]
```diff
@@ -71,15 +66,21 @@ function AdderToolbarArrow({ arrowDirection }) {
...
- *  @param {string|null} props.shortcut
- */
-function ToolbarButton({ badgeCount, Icon, label, onClick, shortcut }) {
+type ToolbarButtonProps = {
+  badgeCount?: number;
+  Icon?: IconComponent;
+  label: string;
+  onClick: () => void;
+  shortcut: string | null;
+};
```
Comment (lyzadanger) [inline src/annotator/components/AdderToolbar.tsx:75]: Just checking in that we still want to alphabetize props, not put optional props last or anything.

File: src/annotator/components/AdderToolbar.tsx [hunk 2]
```diff
@@ -112,25 +113,26 @@ function ToolbarButton({ badgeCount, Icon, label, onClick, shortcut }) {
   );
 }
 
-/**
- * Union of possible toolbar commands.
- *
- * @typedef {'annotate'|'highlight'|'show'|'hide'} Command
- */
+export type Command = 'annotate' | 'highlight' | 'show' | 'hide';
```
Comment (lyzadanger) [inline src/annotator/components/AdderToolbar.tsx:116]: This type is used by `adder`. Do we need to establish a convention about how we want to handle type exports?

File: src/annotator/components/AdderToolbar.tsx (hunk 1 above)
Comment (robertknight) [reply src/annotator/components/AdderToolbar.tsx:75]: I haven't given this much thought. I was planning to mostly preserve the existing ordering when converting code.

File: src/annotator/components/AdderToolbar.tsx (hunk 2 above)
Comment (robertknight) [reply src/annotator/components/AdderToolbar.tsx:116]: If you want to use the type in other modules, or intend for it to be directly used (eg. an interface that we expect classes to declare they implement), then it needs to be exported. We haven't yet established conventions around types that _could_ be used, but aren't often (eg. props types). One possible policy is to export all types which are referenced in the module's value exports (functions, classes, variables).

**Review by robertknight — APPROVED** [review]

**Pull request merged**